supervisor/docker/addon.py

```diff
@@ -636,15 +654,52 @@ def _stop(self, remove_container=True) -> None:
 
         Need run inside executor.
         """
+        # DNS
         if self.ip_address != NO_ADDDRESS:
             try:
                 self.sys_plugins.dns.delete_host(self.addon.hostname)
             except CoreDNSError as err:
                 _LOGGER.warning("Can't update DNS for %s", self.name)
                 self.sys_capture_exception(err)
+
+        # Hardware
+        if self._hw_listener:
+            self.sys_bus.remove_listener(self._hw_listener)
+            self._hw_listener = None
+
         super()._stop(remove_container)
 
     def _validate_trust(
         self, image_id: str, image: str, version: AwesomeVersion
     ) -> None:
         """Validate trust of content."""
+
+    @Job(conditions=[JobCondition.OS_AGENT])
+    async def _hardware_events(self, device: Device) -> None:
+        """Process Hardware events for adjust device access."""
+        if not any(
+            device_path in (device.path, device.sysfs)
+            for device_path in self.addon.static_devices
+        ):
+            return
+
+        try:
+            docker_container = self.sys_docker.containers.get(self.name)
+        except docker.errors.NotFound:
+            self.sys_bus.remove_listener(self._hw_listener)
+            self._hw_listener = None
+            return
+        except (docker.errors.DockerException, requests.RequestException) as err:
+            raise DockerError(
+                f"Can't process Hardware Event on {self.name}: {err!s}", _LOGGER.error
+            ) from err
+
+        try:
+            await self.sys_dbus.agent.cgroup.add_devices_allowed(
+                docker_container.id, self.sys_hardware.policy.get_cgroups_rule(device)
+            )
```

Comment:
Should we maybe add a info message here?
```suggestion
            )
            _LOGGER.info("Added cgroup permissions for device %s to %s", device.path, self.name)
```